### ISSUE TITLE
Access denied to the specified API version

### ISSUE THREAD

**Issue by mkliu** [Azure/azure-xplat-cli#2047]
This happens to multiple operations, such as creating key vault, listing AD service principals. The error is

```
hu Aug 20 2015 00:10:31 GMT-0700 (PDT):
{ [Error: Access denied to the specified API version.]
  code: 'Authentication_Unauthorized',
  statusCode: 403,
  __frame:
   { name: 'getAccountBySPN',
     line: 795,
     file: '/usr/local/lib/node_modules/azure-cli/lib/commands/arm/keyvault/keyvault._js',
     prev:
      { name: 'getAccountFromSubscription',
        line: 825,
        file: '/usr/local/lib/node_modules/azure-cli/lib/commands/arm/keyvault/keyvault._js',
        prev: [Object],
        calls: 1,
        active: false,
        offset: 15,
        col: 15 },
     calls: 0,
     active: false,
     offset: 5,
     col: 48 },
  rawStack: [Getter] }
Error: Access denied to the specified API version.
    at Function.ServiceClient._normalizeError (/usr/local/lib/node_modules/azure-cli/node_modules/azure-common/lib/services/serviceclient.js:815:23)
    at /usr/local/lib/node_modules/azure-cli/node_modules/azure-common/lib/services/filters/errorhandlingfilter.js:44:29
    at Request._callback (/usr/local/lib/node_modules/azure-cli/node_modules/azure-common/lib/http/request-pipeline.js:109:14)
    at Request.self.callback (/usr/local/lib/node_modules/azure-cli/node_modules/azure-common/node_modules/request/request.js:199:22)
    at Request.emit (events.js:110:17)
    at Request.<anonymous> (/usr/local/lib/node_modules/azure-cli/node_modules/azure-common/node_modules/request/request.js:1160:14)
    at Request.emit (events.js:129:20)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/azure-cli/node_modules/azure-common/node_modules/request/request.js:1111:12)
    at IncomingMessage.emit (events.js:129:20)
    at _stream_readable.js:908:16
```

I'm on tool version 0.9.7. Login via service principal coz 2FA isn't supported here.

**Comment by amarzavery** [comment]
@mkliu - Can you execute ```azure account clear -q``` and login again as a service principal.
- After logging in as an SP, if you execute ```azure account list``` do you see the list of subscriptions associated with this SP?
- Does a simple command like ```azure group list``` work. Note you need to be in ARM mode to execute this command. (```azure config mode arm```).

**Comment by mkliu** [comment]
@amarzavery i tried again on a clean computer, still the same thing.
azure account list works, and azure group list works too.

**Comment by amarzavery** [comment]
@mkliu this issue should be fixed in #2944. We have moved to a public api-version 1.6 of the Graph API. So now, after successful login as an SP, you should be able to perform the desired operations.
